lemur/common/validators.py

```diff
@@ -41,22 +42,33 @@ def private_key(key):
         raise ValidationError('Private key presented is not valid.')
 
 
+def common_name(value):
+    """If the common name could be a domain name, apply domain validation rules."""
+    # Common name could be a domain name, or a human-readable name of the subject (often used in CA names or client
+    # certificates). As a simple heuristic, we assume that human-readable names always include a space.
+    # However, to avoid confusion for humans, we also don't count spaces at the beginning or end of the string.
```

Comment:
The problem with your proposal is that `myCommonName` or `com` is also a valid domain name, and if implemented that way, it would be impossible to restrict that. In a corporate setting, where workstations have configured a domain search path, simple strings like that could resolve to an actual service.

My point was to err on the side of caution: only skip whitelist check if it's clear that it couldn't be a domain name.